pkgs/development/tools/glualint/default.nix

```
@@ -0,0 +1,104 @@
+{ mkDerivation
+, aeson
+, array
+, base
+, bytestring
+, containers
+, directory
+, deepseq
+, filemanip
+, filepath
+, ListLike
+, MissingH
+, mtl
+, optparse-applicative
+, parsec
+, pretty
+, signal
+, lib
+, uu-parsinglib
+, uuagc
+, uuagc-cabal
+, vector
+, pkgs
+, fetchFromGitHub
+}:
+mkDerivation {
+  # Clean up the source of the derivation to prevent rebuilds
+  src = lib.cleanSourceWith {
+    src = fetchFromGitHub {
+      owner = "FPtje";
+      repo = "GLuaFixer";
+      rev = "1.24.1";
+      hash = "sha256-zL0GcrNihSOaw5JkDi04ipOmBq8idj2m0VCKU2J1ZbA=";
+    };
+    filter = path: type:
+      let
+        suffixAllowlist = [
+          ".ag"
+          ".cabal"
+          ".hs"
+          "AGGenerator.sh"
+          "LICENSE"
+        ];
+        suffixDenylist = [
+          "release-linux.sh"
+        ];
+      in
+      ((type == "directory") ||
+      (builtins.any (suffix: lib.hasSuffix suffix path) suffixAllowlist)
+      ) &&
+      !(builtins.any (suffix: lib.hasSuffix suffix path) suffixDenylist) &&
+      # Simple library function to remove git related files.
+      lib.cleanSourceFilter path type
+    ;
+  };
```

Would require ifd which kind of defeats the point and is forbidden in nixpkgs.

```suggestion
  src = fetchFromGitHub {
    owner = "FPtje";
    repo = "GLuaFixer";
    rev = "1.24.1";
    hash = "sha256-zL0GcrNihSOaw5JkDi04ipOmBq8idj2m0VCKU2J1ZbA=";
  };
```